### DISCUSSION TITLE
Is it possible to use vi motions in the default internal shell?

### DISCUSSION THREAD

**Discussion by cohml** [inducer/pudb#633] [Q&A]
The native Python REPL uses readline, meaning configurations set in `~/.inputrc` are respected. One such configuration is `set editing-mode vi`, which enables some basic vi motions.

`pudb`'s "internal shell" seems exactly like the native Python REPL, so I had suspected it would use readline as well. However, I am unable to use vi motions in the internal shell, despite `set editing-mode vi` being enumerated in my `~/.inputrc`.

So how, if at all, can I use vi motions in `pudb`'s internal shell? I don't see that this is addressed in the README/documentation.

**Comment by cohml** [comment]
Possibly related to ##547.

**Comment by inducer** [comment]
That text entry box uses https://github.com/rr-/urwid_readline, so probably the most useful course of action would be to extend that to understand vi keys. It's not a priority for me, but if some were to put together PRs to make it happen, I likely wouldn't stand in the way.
